Introduce `publish-library-convention`.

atrafimchyk opened the pull request.

Commit:
Introduce `publish-library-convention`.

rasskazovaleksey commented on build-logic/src/main/kotlin/publish-library-convention.gradle.kts:
```diff
@@ -0,0 +1,50 @@
+project.version = project.version.toString() + versionSuffix
+
+mavenPublishing {
+    publishToMavenCentral(com.vanniktech.maven.publish.SonatypeHost.S01)
```
[Not critical] Readability. You can use import.

rasskazovaleksey commented on build-logic/src/main/kotlin/publish-library-convention.gradle.kts:
```diff
@@ -0,0 +1,50 @@
+    pom {
+        name.set("humans-mvi")
+        description.set("Simple and concise implementation of Redux/MVI approach by Humans.")
+        inceptionYear.set("2023")
```
[Improvement] Hardcoded year. This will break in 2024 and will require manual update. Remove or https://stackoverflow.com/questions/136419/get-integer-value-of-the-current-year-in-java

atrafimchyk replied:
I'm not sure that you've got it right.
according to the documentation:
```
The year of the inception for the publication represented by this POM
```

so, we don't need to change this property next year.

rasskazovaleksey commented on mvi-core/build.gradle.kts:
```diff
@@ -1,13 +1,16 @@
 
+ext["artifactId"] = "mvi-core"
+
+apply(plugin = "publish-library-convention")
```
[Improvement] Convention shall be applied to plugins block.

rasskazovaleksey commented on build-logic/src/main/kotlin/publish-library-convention.gradle.kts:
```diff
@@ -0,0 +1,50 @@
+
+    coordinates(
+        groupId = project.group.toString(),
+        artifactId = project.ext["artifactId"].toString(),
```
[Improvement] no need to use ext map. ext["artifactId"] = "mvi-core" in lib project can be removed. This is actually project.name which is set to projects folder if not set directly.

atrafimchyk replied:
thx, will check it

rasskazovaleksey reviewed (changes requested):
Good job. Small improvements can be done.

Commit:
Code review fixes:
- import constant
- use `project.name` as `artifactId`
- simplify plugin configuration.
